client/src/modules/stock/lots-duplicates/duplicate-lots.js

```diff
@@ -0,0 +1,165 @@
+  vm.enterprise = Session.enterprise;
+
+  // const duplicateLotsFilters = Stock.filter.inventory;
+
+  const columns = [
+    {
+      field : 'label',
+      displayName : 'TABLE.COLUMNS.LABEL',
+      headerTooltip : 'TABLE.COLUMNS.LABEL',
+      headerCellFilter : 'translate',
+
+    }, {
+      field : 'inventory_text',
+      displayName : 'TABLE.COLUMNS.INVENTORY',
+      headerTooltip : 'TABLE.COLUMNS.INVENTORY',
+      headerCellFilter : 'translate',
+      width : '20%',
+    }, {
+      field : 'quantity',
+      displayName : 'TABLE.COLUMNS.QUANTITY',
+      headerTooltip : 'TABLE.COLUMNS.QUANTITY',
+      headerCellFilter : 'translate',
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'initial_quantity',
+      displayName : 'STOCK.INITIAL_QUANTITY',
+      headerTooltip : 'STOCK.INITIAL_QUANTITY',
+      headerCellFilter : 'translate',
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'unit_cost',
+      displayName : 'TABLE.COLUMNS.UNIT_PRICE',
+      headerTooltip : 'TABLE.COLUMNS.UNIT_PRICE',
+      headerCellFilter : 'translate',
+      cellFilter : `currency:${vm.enterprise.currency_id}`,
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'entry_date',
+      displayName : 'STOCK.ENTRY_DATE',
+      headerTooltip : 'STOCK.ENTRY_DATE',
+      headerCellFilter : 'translate',
+      cellFilter : 'date',
+      cellClass : 'text-right',
+    }, {
+      field : 'expiration_date',
+      displayName : 'STOCK.EXPIRATION_DATE',
+      headerTooltip : 'STOCK.EXPIRATION_DATE',
+      headerCellFilter : 'translate',
+      cellFilter : 'date:"mediumDate"',
+      cellClass : 'text-right',
+    }, {
+      field : 'num_duplicates',
+      displayName : 'TABLE.COLUMNS.NUM_DUPLICATE_LOTS',
+      headerTooltip : 'TABLE.COLUMNS.NUM_DUPLICATE_LOTS',
+      headerCellFilter : 'translate',
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'action',
+      displayName : '',
+      enableFiltering : false,
+      enableSorting : false,
+      cellTemplate : 'modules/stock/lots-duplicates/templates/action.cell.html',
+    },
+  ];
+
+  const footerTemplate = `
+    <div class="ui-grid-cell-contents">
+      <b>{{ grid.appScope.gridOptions.data.length }}</b>
+      <span translate>FORM.INFO.FOUND</span>
+    </div>
+  `;
+
+  vm.gridOptions = {
+    appScopeProvider   : vm,
+    enableColumnMenus  : false,
+    columnDefs         : columns,
+    showGridFooter     : true,
+    gridFooterTemplate : footerTemplate,
+    fastWatch          : true,
+    flatEntityAccess   : true,
+    onRegisterApi,
+  };
+
+  const gridColumns = new Columns(vm.gridOptions, cacheKey);
+  const state = new GridState(vm.gridOptions, cacheKey);
+
+  vm.gridApi = {};
+  vm.saveGridState = state.saveGridState;
+
+  function clearGridState() {
+    state.clearGridState();
+    $state.reload();
+  }
+
+  function onRegisterApi(gridApi) {
```

Comment:
You can probably remove all references to `gridApi`.  I don' think you need it or use it anywhere.  I can't see a scenario where you would need it either.

Reply:
Removed.